### PR TITLE
addpkg: shadowsocks-rust-git

### LINKED ISSUE #2206: Add shadowsocks-rust
### 问题类型 / Type of issues

* 软件打包请求 / package request

----

* 简述 / Brief Description

    shadowsocks-rust

* 主要语言和工具链 / Main Languages and Toolchains

    rust

* 上游链接 / Upstream Link

    [https://github.com/shadowsocks/shadowsocks-rust](https://github.com/shadowsocks/shadowsocks-rust)

* 为何需要这个软件包 / Why Need This Package

    Due to [shadowsocks-libev](https://github.com/shadowsocks/shadowsocks-libev) entered bug-fix-only status. Upstream recommends this implementation.

### PR DESCRIPTION
**Pull request by DuckSoft**
This fixes #2206.

### CODE REVIEW

File: archlinuxcn/shadowsocks-rust-git/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ 	install -Dm644 LICENSE                             -vt "$pkgdir"/usr/share/licenses/shadowsocks-rust/
+	install -Dm644 examples/config{,_ext}.json         -vt "$pkgdir"/etc/shadowsocks-rust/
+	
+ 	cd target/release
+ 	for name in sslocal ssserver ssurl ssmanager; do
+        install -Dm755 -v $name "$pkgdir"/usr/bin/${name}-rust
+    done
+}
+
+b2sums=('SKIP'
```
Comment (imlonghao) [inline archlinuxcn/shadowsocks-rust-git/PKGBUILD:39]: what is it? `b2sums`?

Comment (DuckSoft) [reply archlinuxcn/shadowsocks-rust-git/PKGBUILD:39]: https://en.wikipedia.org/wiki/BLAKE_(hash_function)

Comment (imlonghao) [reply archlinuxcn/shadowsocks-rust-git/PKGBUILD:39]: Well then, learn something new. :)

File: archlinuxcn/shadowsocks-rust-git/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+pkgver=1.10.9.r5.g89b329c
+pkgrel=1
+provides=(shadowsocks-rust)
+conflicts=(shadowsocks-rust)
+pkgdesc='Rust port of Shadowsocks'
+arch=('x86_64')
+url='https://github.com/shadowsocks/shadowsocks-rust'
+license=('MIT')
+depends=('openssl')
+makedepends=('cargo' 'libsodium')
```
Comment (imlonghao) [inline archlinuxcn/shadowsocks-rust-git/PKGBUILD:13]:
```
==> ERROR: Cannot find the git package needed to handle git sources.
==> ERROR: Build failed, check /var/lib/archbuild/extra-x86_64/imlonghao/build
```

Comment (DuckSoft) [reply archlinuxcn/shadowsocks-rust-git/PKGBUILD:13]: oops, I missed that when converting `shadowsocks-rust` to `-git`...
reverting...

Comment (DuckSoft) [reply archlinuxcn/shadowsocks-rust-git/PKGBUILD:13]: resolved in https://github.com/archlinuxcn/repo/commit/b1e9033118f9ddf8d70df1606bed76f14f9f5471

**Comment by DuckSoft** [comment]
Additional Info: I would also like to package `shadowsocks-rust-opt-git`, which is basically the same, but targets `x86-64-v3` platform and will make use of modern processor instruction sets (like AVX & AVX2).

**Pull request merged**